layout.ld

```diff
@@ -18,6 +18,10 @@
  *     INCLUDE ../libtock-rs/layout.ld
  */
 
+/*
+ * Any change to STACK_SIZE should be accompanied by a corresponding change to
+ * `elf2tab`'s `--stack` option
+ */
 STACK_SIZE = 2048;
```

Comment:
You don't need to change this in this CL but I want to point out that STACK_SIZE should probably be set by the application-specific linker file rather than this linker file.

Reply:
Moving `STACK_SIZE` to application-specific linker file is a good idea. I'll do it in V2 of this patch series.

I think we do need to change `elf2tab`'s `--stack` option when we change `STACK_SIZE`. The reason for the dependency between `STACK_SIZE` and `elf2tab`'s `--stack` option is as follows.

1. `STACK_SIZE` in `layout.ld` is used to only determine the values in `.crt0_header` (`LayoutHeader`). `elf2tab` is not aware of what `STACK_SIZE` is used by linker file

2. When `elf2tab` uses `--stack` option, it is using it to determine `minimum_application_size` for the TBF header.

3. When the kernel loads the application, it uses `minimum_application_size` to tell the MPU trait the minimum amount of unprivileged memory to allocate for the process. The kernel is not aware of where in `minimum_application_size` the stack would be placed. It just sets it to `mem_start + 3072` and expects `_start` to set the correct stack pointer based on `STACK_SIZE`.

If we were to change `STACK_SIZE` to lets say 16KB, and not make corresponding changes with `elf2tab`'s `--stack` option, then `min_application_size` would be wrong, so the MPU region would be smaller than what we will need. Then in `_start`, when we set the stack pointer, it will be outside the range of unprivileged memory resulting in a memory fault when we call `rust_start`.

Reply:
The dependency with elf2tab is exactly why I think the application should do it: the application's build script (e.g. Makefile) should be calling elf2tab already.